guides/source/sign_up_and_settings.md

```diff
@@ -0,0 +1,1658 @@
+**DO NOT READ THIS FILE ON GITHUB, GUIDES ARE PUBLISHED ON <https://guides.rubyonrails.org>.**
+
+Sign Up and Settings
+========================
+
+This guide covers adding Sign Up and Settings to the store e-commerce application in the [Getting Started Guide](getting_started.html)). We will use the final code from that guide as a starting place.
+
+After reading this guide, you will know how to:
+
+* Add user Sign Up
+* Rate limit controller actions
+* Create a nested layout
+* Separate controllers by role (users and admins)
+* Write tests for users with different roles
+
+--------------------------------------------------------------------------------
+
+Introduction
+------------
+
+One of the most common features to add to any application is a sign up process for registering new users. The e-commerce application we've built so far only has authentication and users must be created in the Rails console or a script.
+
+This feature is required before we can add other features. For example, to let users create wishlists, they will need to be able to sign up first before they can create a wishlist associated with their account.
+
+Let's get started!
+
+Adding Sign Up
+--------------
+
+We've already used the Rails authentication generator to allow users to login to their accounts. The generator created a `User` model with `email_address:string` and `password_digest:string` columns in the database. It also added `has_secure_password` to the `User` model which handles passwords and confirmations. This takes care of most of what we need to add sign up to our application.
+
+### Adding Names To Users
+
+It's also a good idea to collect the user's name at sign up. This allows us to personalize their experience and address them directly in the application. Let's start by adding `first_name` and `last_name` columns to the database.
```

Comment:
Is the first/last name separation relevant for this guide? The autocomplete attributes in the form are cool, but in general I think storing those fields separately can be annoying for users. Outside of the US/UK is fairly common for people to have several names, so I often don't know where to put those. Is not my first (given) name nor the last one.

It seems that we always use the full name, so we could just store that.

I don't meant to rant here, I just think the simplification helps with the UX _and_ makes this guide simpler/shorter.

Reply:
Nice. Disregard then. TY!